### ISSUE TITLE
Rethink the UX of the Enhanced Session Recording guide

### ISSUE THREAD

**Issue by ptgott** [gravitational/teleport#15311]
## Details

Since Enhanced Session Recording comes with some critical security risks that we've [begun to document](https://github.com/gravitational/teleport/pull/15191/files) within the guide, we should rethink the way users work through the guide's instructions. 

The list of caveats for using Enhanced Session Recording is fairly extensive, and the current wording provides a broad indication of the precautions an organization should have in place (e.g., "Proper network-based restrictions for ingress traffic") that implies a high level of skill and knowledge on the part of the user, as well as some familiarity with how Enhanced Session Recording works. 

If we expect users to adopt Enhanced Session Recording on their own, we should include a more comprehensive, verifiable set of steps for setting up a user's environment so they can use this feature securely. Otherwise, users will need to understand Enhanced Session Recording well enough to implement specific defenses, and I'm not sure the current guide is up to the task.

If we expect users to set up Enhanced Session Recording only after conversations with Teleport sales engineering and support staff where they can be confident in having configured their systems securely, we should make that more explicit.

What do you think, @xinding33? Also, would there be a way to have a `teleport` binary determine whether the required security precautions are met for Enhanced Session Recording when it first loads a config that enables this feature? (Maybe that's a separate issue.)

Also CC @reedloden 

### Category

- Improve Existing

**Comment by ptgott** [comment]
Closing. This is not a high-traffic page, and it is probably fine for users who want to implement the feature to carefully read a screen or two worth of detailed information.
